csharp/CMakeLists.txt

```diff
@@ -265,8 +235,25 @@ add_custom_target(csharp_sdk ALL
 )
 
 if( BUILD_NUGET_PACKAGE )
+  if(NOT CMAKE_CL_64)
+    message(FATAL_ERROR "Nuget package cannot be built using the 64 bit generator")
+  endif()
 
   if (MSVC)
+    include(ExternalProject)
+    
+    ExternalProject_Add(OS-32
```

Comment:
@tijcolem this is the root of the magic that enables universal nuget packages. We generate and compile an entirely new build tree, which is configured for 32 bit. Might want to update the CMAKE_GENERATOR to use ninja, although I am not sure off hand how to identify the 32 bit generator for ninja on windows.